libarchive/archive_read_support_format_iso9660.c

```diff
@@ -1763,7 +1764,11 @@ parse_file_info(struct archive_read *a, struct file_info *parent,
 
 	iso9660 = (struct iso9660 *)(a->format->data);
 
-	dr_len = (size_t)isodirrec[DR_length_offset];
+	if (reclen == 0 || reclen < (dr_len = (size_t)isodirrec[DR_length_offset])) {
```

Comment:
I'd suggest leaving the `dr_len = ...;` logic on its own line for clarity.  And for some reason, the condition `dr_len > reclen` makes it more clear to me that we're checking for overflow, but maybe that's just the long line getting to me.  (Style nit: the preexisting cast from `unsigned char` to `size_t` is unnecessary in C.)  Perhaps:

```
if (reclen != 0)
    dr_len = isodirrec[DR_length_offset];
if (reclen == 0 || dr_len > reclen) {
    archive_set_error(...);
    return (NULL);
}
```

However, your version is functionally correct.